addon/components/nav-action.js

```diff
@@ -1,44 +1,43 @@
 import Ember from 'ember'
+import {Component} from 'ember-frost-core'
+import {PropTypes} from 'ember-prop-types'
 import layout from '../templates/components/nav-action'
-import PropTypeMixin, {PropTypes} from 'ember-prop-types'
 
 const {
-  Component,
-  get,
   inject: {
     service
   }
 } = Ember
-export default Component.extend(PropTypeMixin, {
+export default Component.extend({
   // == Services ==============================================================
 
   frostNavigation: service(),
 
   // == Component properties ==================================================
-
-  classNames: [
-    'nav-route',
-    'nav-action'
-  ],
   layout,
 
   // == State properties ======================================================
 
   propTypes: {
-    item: PropTypes.object.isRequired
+    item: PropTypes.object.isRequired,
+    hook: PropTypes.string
   },
+
   getDefaultProps () {
     return {
       item: {
         pack: 'frost'
-      }
+      },
+      hook: 'nav-action'
```

Comment:
A default was fine for the overall navigation (it's unique to an app), but here we want the consumer to provide their own and get a warning (from propTypes) if they don't, since each action should be unique